contracts/messagebuscontract/IMessageBus.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./Structs.sol";
+
+interface IMessageBus {
+    // This method is called from contracts to publish messages to the other linked message bus.
+    // nonce - This is provided and serves as deduplication nonce. It can also be used to group a batch of messages together.
+    // topic - This is the topic for which the payload is published. 
+    // payload - This is the actual message.
+    // consistencyLevel - this is how many block confirmations to wait before publishing the message. 
+    // Notice that consistencyLevel == 0 is still secure, but might make your protocol result more prone to reorganizations.
+    // returns sequence - this is the unique id of the published message for the address calling the function. It can be used
+    // to determine the order of incoming messages on the other side and if something is missing.
+    function publishMessage(
+        uint32 nonce,
+        uint32 topic,
+        bytes calldata payload, 
+        uint8 consistencyLevel
+    ) external returns (uint64 sequence);
+
+    // This function verifies that a cross chain message provided by the caller has indeed been submitted from the other network
+    // and returns true only if the challenge period for the message has passed.
+    function verifyMessageFinalized(Structs.CrossChainMessage calldata crossChainMessage) external view returns (bool);
+    
+    // Returns the time when a message is final (when the rollup challenge period has passed). If the message was never submitted the call will revert.
+    function getMessageTimeOfFinality(Structs.CrossChainMessage calldata crossChainMessage) external view returns (uint256);
```

Comment:
why can't this receive the hash?
That would avoid having to compute the hash inside this function

Reply:
It can receive a hash, but I'm not sure if it is the correct approach to go forward. Receiving a hash would mean the calling smart contract would have to do the hashing as you usually need the full message to perform so logic like for example when the message is encoded structure for transfering assets. I guess making it a hash would reduce gas costs, but overall I'm not sure if this approach is good UX.

Reply:
ok. but let's keep this in mind.